agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4398,10 +4281,210 @@ bool slave_thread::send_cmdu_to_controller(ieee1905_1::CmduMessageTx &cmdu_tx)
+    // the destination is our AL-MAC, and the controller does listen to this CMDU.
+    // Ideally, we should use the message type attribute from the WSC payload to distinguish.
+    // Unfortunately, that is a bit complicated with the current tlv parser. However, there is another
+    // way to distinguish them: the M1 message has the AP_Radio_Basic_Capabilities TLV,
+    // while the M2 has the AP_Radio_Identifier TLV.
+    // If this is a looped back M2 CMDU, we can treat is as handled successfully.
+    if (cmdu_rx.getNextTlvType() == wfa_map::TLV_AP_RADIO_BASIC_CAPABILITIES)
+        return true;
+
+    /**
+    * @brief Parse AP-Autoconfiguration WSC which should include one AP Radio Identifier
+    * TLV and one or more WSC TLV containing M2
+    */
+    auto ruid = cmdu_rx.addClass<wfa_map::tlvApRadioIdentifier>();
+    if (ruid == nullptr) {
+        LOG(ERROR) << "Failed to get tlvApRadioIdentifier TLV";
+        return false;
+    }
+
+    // Check if the message is for this radio agent by comparing the ruid
+    if (!config.radio_identifier.compare(network_utils::mac_to_string(ruid->radio_uid())))
+        return true;
+
+    LOG(DEBUG) << "Received AP_AUTOCONFIGURATION_WIFI_SIMPLE_CONFIGURATION_MESSAGE";
+    // parse all M1 TLVs
+    std::vector<std::shared_ptr<ieee1905_1::tlvWscM2>> m2_list;
+    while (1) {
+        if (cmdu_rx.getNextTlvType() != uint8_t(ieee1905_1::eTlvType::TLV_WSC))
+            break;
+
+        auto m2 = cmdu_rx.addClass<ieee1905_1::tlvWscM2>();
+        if (!m2) {
+            LOG(ERROR) << "Not an WSC M2 TLV!";
+            return false;
+        }
+        m2_list.push_back(m2);
+    }
+    if (m2_list.empty()) {
+        LOG(ERROR) << "No M2 TLVs present";
+        return false;
+    }
+    //TODO parse all of the M2 TLVs
+    auto m2                  = m2_list[0];
+    const std::string intel  = "Intel";
+    std::string manufacturer = std::string(m2->M2Frame().manufacturer_attr.data,
+                                           m2->M2Frame().manufacturer_attr.data_length);
+    if (!manufacturer.compare(intel)) {
+        //TODO add support for none Intel agents
+        LOG(ERROR) << "None Intel controller " << manufacturer << " , dropping message";
+        return false;
+    }
+    LOG(INFO) << "Intel controller join response";
+    if (!parse_intel_join_response(sd, cmdu_rx)) {
+        LOG(ERROR) << "Parse join response failed";
+        return false;
+    }
+
+    return true;
+}
+
+bool slave_thread::parse_intel_join_response(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
 {
-    LOG(INFO) << "received CHANNEL_PREFERENCE_QUERY_MESSAGE";
+    LOG(DEBUG) << "ACTION_CONTROL_SLAVE_JOINED_RESPONSE sd=" << intptr_t(sd);
+    if (slave_state != STATE_WAIT_FOR_JOINED_RESPONSE) {
+        LOG(ERROR) << "slave_state != STATE_WAIT_FOR_JOINED_RESPONSE";
+        return false;
+    }
+
+    if (cmdu_rx.getNextTlvType() != uint8_t(ieee1905_1::eTlvType::TLV_VENDOR_SPECIFIC)) {
```

Comment:
1. maybe should be done outside the handle function?
2. Shouldn't you also check for OUI?

Reply:
1. OK
2. Not needed, we check it later in `parse_intel_vs_message()`